include/aspect/traction_boundary_conditions/interface.h

```diff
@@ -0,0 +1,203 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__traction_boundary_conditions_interface_h
+#define __aspect__traction_boundary_conditions_interface_h
+
+#include <aspect/plugins.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/point.h>
+#include <deal.II/base/parameter_handler.h>
+
+namespace aspect
+{
+  /**
+   * A namespace in which we define everything that has to do with defining
+   * traction boundary boundary conditions for the Stokes equations.
+   *
+   * @ingroup TractionBoundaryConditionsModels
+   */
+  namespace TractionBoundaryConditions
+  {
+    using namespace dealii;
+
+    /**
+     * A base class for parameterizations of traction boundary conditions.
+     *
+     * @ingroup TractionBoundaryConditionsModels
+     */
+    template <int dim>
+    class Interface
+    {
+      public:
+        /**
+         * Destructor. Made virtual to enforce that derived classes also have
+         * virtual destructors.
+         */
+        virtual ~Interface();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program after parse_parameters is run and after the
+         * SimulatorAccess (if applicable) is initialized.
+         */
+        virtual
+        void
+        initialize ();
+
+        /**
+         * A function that is called at the beginning of each time step.
+         * The default implementation of the function does nothing, but
+         * derived classes that need more elaborate setups for a given time
+         * step may overload the function.
+         *
+         * The point of this function is to allow complex boundary traction
+         * models to do an initialization step once at the beginning of each
+         * time step. An example would be a model that needs to call an
+         * external program to compute stresses for a set of plates.
+         */
+        virtual
+        void
+        update ();
+
+        /**
+         * Return the boundary traction as a function of position. The
+         * (outward) normal vector to the domain is also provided as
+         * a second argument.
+         */
+        virtual
+        Tensor<1,dim>
+        traction (const Point<dim> &position,
+                  const Tensor<1,dim> &normal_vector) const = 0;
+
+        /**
+         * Declare the parameters this class takes through input files. The
+         * default implementation of this function does not describe any
+         * parameters. Consequently, derived classes do not have to overload
+         * this function if they do not take any runtime parameters.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
```

Comment:
You will get warnings about prm not used with newer deal.II versions. Please remove just the name 'prm' from this line or comment it out (ParameterHandler &);